teos/src/gatekeeper.rs

```diff
@@ -134,8 +116,17 @@ impl Gatekeeper {
     }
 
     /// Gets the data held by the tower about a given user.
-    pub(crate) fn get_user_info(&self, user_id: UserId) -> Option<UserInfo> {
-        self.registered_users.lock().unwrap().get(&user_id).cloned()
+    pub(crate) fn get_user_info(&self, user_id: UserId) -> Option<(UserInfo, Vec<Locator>)> {
+        let info = self.registered_users.lock().unwrap().get(&user_id).cloned();
```

Comment:
You mean why not chaining `.cloned()` with `.map(|...`?
I assumed this would make it more readable.
\+ actually now that you have said it, `load_user_locators` is indeed expected to take some good IO time.